src/hotspot/share/gc/parallel/psParallelCompact.cpp

```diff
@@ -1156,6 +1157,38 @@ static void flush_marking_stats_cache(const uint num_workers) {
   }
 }
 
+class PSParallelCleaningTask : public WorkerTask {
+  bool                    _unloading_occurred;
+  CodeCacheUnloadingTask  _code_cache_task;
+  // Prune dead klasses from subklass/sibling/implementor lists.
+  KlassCleaningTask       _klass_cleaning_task;
+
+public:
+  PSParallelCleaningTask(bool unloading_occurred) :
+    WorkerTask("PS Parallel Cleaning"),
+    _unloading_occurred(unloading_occurred),
+    _code_cache_task(unloading_occurred),
+    _klass_cleaning_task() {}
+
+  void work(uint worker_id) {
+    if (worker_id == 0) {
+      // Serial work; only first worker.
+      // Clean JVMCI metadata handles.
+      JVMCI_ONLY(JVMCI::do_unloading(_unloading_occurred));
```

Comment:
Potentially do not execute this if `EnableJVMCI` is false. Probably does not change a lot.